v2/spiffetls/dial_test.go

```diff
@@ -253,26 +271,33 @@ func TestDialWithMode(t *testing.T) {
 			}()
 
 			// Assertions
+		loop:
 			for {
 				select {
 				case dialConn := <-dialConnCh:
 					require.NotNil(t, dialConn)
 					defer dialConn.Close()
 
+					if test.usesExternalDialer {
+						require.True(t, externalDialerUsed)
+					}
+					if test.usesBaseTLSConfig {
+						require.NotEmpty(t, externalTLSConfBuffer.String())
```

Comment:
nit: really not a big deal but there is no reason to get the string. NotEmpty will also assert it isn't a zero so we can just check length.
```suggestion
						require.NotEmpty(t, externalTLSConfBuffer.Len())
```

Reply:
Right, `require.NotEmpty(t, externalTLSConfBuffer)` works too

Reply:
Almost, it will check to see if the buffer is equal to the zero value, which will work in this case.

However, it is a little more nuanced because if you created a buffer and then drained it (with Read() calls for example), even if Len() returned 0, if you did `require.NotEmpty(t, buf)`, the assertion would pass.